### PR TITLE
Issue #45: Don't delete config.json

### PR DESCRIPTION
**Pull request by piebrain**

### CODE REVIEW

File: lib/Compass/Settings.cpp [hunk 1]
```diff
@@ -1,25 +1,50 @@
...
     }
 }
 
+long SettingsClass::getSavedTimestamp() {
+    String tFilename = F("/." SETTINGS_CONFIG_FILE ".timestamp");
+    if(SDFS.exists(tFilename)) {
+        File tSD = SDFS.open(tFilename, "r");
+        time_t tSDts = tSD.readStringUntil('\n').toInt();
+        tSD.close();
+        return tSDts;
```
Comment (sastel) [inline lib/Compass/Settings.cpp:34]: I'm not a fan of using two return statements in this case. 
Please replace with a line at the top that puts "time_t tSDts = 0" on the stack, set tSDts = x in here, and then return tSDts at the end.

File: lib/Compass/Settings.cpp [hunk 2]
```diff
@@ -1,25 +1,50 @@
 #include "Settings.h"
 
 void SettingsClass::copyJsonToFlash() {
-    String cFilename = F(SETTINGS_CONFIG_FILE);
+    String cFilename = F("/" SETTINGS_CONFIG_FILE);
     if(SDFS.exists(cFilename)) {
         File cSD = SDFS.open(cFilename, "r");
-        File cFlash = LittleFS.open(cFilename, "w");
+        time_t cSDts = cSD.getLastWrite();
```
Comment (sastel) [inline lib/Compass/Settings.cpp:7]: How long until this rolls over?

Comment (sastel) [reply lib/Compass/Settings.cpp:7]: Or is it the time_t from time.h which is the number of seconds since the Epoch? In that case we're fine.

**Review by sastel — APPROVED** [review]

File: lib/Compass/Settings.cpp (hunk 2 above)
Comment (piebrain) [reply lib/Compass/Settings.cpp:7]: It reads modify time form the file which should be written by the computer changing the file. It is a unix timestamp I believe.

**Pull request merged**